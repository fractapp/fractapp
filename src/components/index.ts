export {BlueButton} from './BlueButton';
export {WhiteButton, Img} from './WhiteButton';
export {SeedButton} from './SeedButton';
export {StatisticsBar} from './StatisticsBar';
export {PasswordInput} from './PasswordInput';
export {TextInput} from './TextInput';
export {TransactionInfo} from './TransactionInfo';
export {WalletInfo} from './WalletInfo';
export {WalletDetailsInfo} from './WalletDetailsInfo';
export {Dialog} from './Dialog';
export {Loader} from './Loader';
export {DriveItemView} from './DriveItemView';
export {PassCode} from './PassCode';
export {SuccessButton} from './SuccessButton';
export {WalletLogo} from './WalletLogo';
export {TabMenu} from './TabMenu';
export {ChatShortInfo} from './ChatShortInfo';
export {PaymentMsg} from './PaymentMsg';
export {Contact} from './Contact';
export {Receiver} from './Receiver';
export {AmountInput} from './AmountInput';
export {AmountValue} from './AmountValue';
export {ReceiverWithEnterAddress} from './ReceiverWithEnterAddress';
