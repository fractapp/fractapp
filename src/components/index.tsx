export {BlueButton} from './BlueButton';
export {WhiteButton, Img} from './WhiteButton';
export {SeedButton} from './SeedButton';
export {StatisticsBar} from './StatisticsBar';
export {PasswordInput} from './PasswordInput';
export {TextInput} from './TextInput';
export {TransactionInfo} from './TransactionInfo';
export {WalletInfo} from './WalletInfo';
export {WalletDetailsInfo} from './WalletDetailsInfo';
export {Dialog} from './Dialog';
export {Loader} from './Loader';
export {DriveItemView} from './DriveItemView';
export {PassCode} from './PassCode';
export {SuccessButton} from './SuccessButton';
export {WalletLogo} from './WalletLogo';
