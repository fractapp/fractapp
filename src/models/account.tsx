import { Keyring } from '@polkadot/api';
import { KeyringPair } from '@polkadot/keyring/types';



/*
const keyring = new Keyring({ type: "sr25519" })

interface CurrencyAdapter {
    address(): string;
}

class PolkadotAndCusamaAdapter implements CurrencyAdapter {
    keyPair: KeyringPair
    constructor(seed: string, currency: Currency) {
        switch (currency) {
            case Currency.Polkadot:
                keyring.setSS58Format(0)
                break;
            case Currency.Kusama:
                keyring.setSS58Format(0)
                break;
        }
        this.keyPair = keyring.addFromUri(seed)
    }

    address() {
        return this.keyPair.address
    }
}

export class Account {
    seed: string;
    adapters: Map<Currency, CurrencyAdapter>

    constructor(seed: string, adapters?: Map<Currency, CurrencyAdapter>) {
        this.seed = seed;
        if (adapters == null) {
            this.adapters = new Map<Currency, CurrencyAdapter>(
                [
                    [Currency.Polkadot, new PolkadotAndCusamaAdapter(seed, Currency.Polkadot)],
                    [Currency.Kusama, new PolkadotAndCusamaAdapter(seed, Currency.Kusama)]
                ]
            )
        } else {
            this.adapters = adapters;
        }
    }


    address(currency: Currency) {
        adapters
    }


    backupAccount(password: String) {

    }
}
*/