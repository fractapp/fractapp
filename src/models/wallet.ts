import { ImageSourcePropType } from "react-native";

export enum Currency {
    Polkadot = 0,
    Kusama
}

export function getLogo(currency: Currency) {
    let logo: ImageSourcePropType
    switch (currency) {
        case Currency.Polkadot:
            logo = require("assets/img/dot.png")
            break
        case Currency.Kusama:
            logo = require("assets/img/kusama.png")
            break
        default:
            throw ("invalid currency")
    }
    return logo
}
export function getSymbol(currency: Currency) {
    let symbol = ""
    switch (currency) {
        case Currency.Polkadot:
            symbol = "DOT"
            break
        case Currency.Kusama:
            symbol = "KSM"
            break
        default:
            throw ("invalid currency")
    }
    return symbol
}

export class Wallet {
    name: string;
    address: string;
    balance: number;
    currency: Currency;
    usdValue: number;

    constructor(name: string, address: string, currency: Currency, balance: number, price: number) {
        this.name = name;
        this.address = address;
        this.balance = balance;
        this.currency = currency;

        this.usdValue = +(balance * price).toFixed(2)
    }
}
