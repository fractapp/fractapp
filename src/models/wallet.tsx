import { ImageSourcePropType } from "react-native";

export enum Currency {
    Polkadot = 0,
    Kusama
}

export class Wallet {
    name: string;
    address: string;
    balance: number;
    currency: Currency;

    symbol: string;
    usdValue: number;
    logo: ImageSourcePropType;

    constructor(name: string, address: string, currency: Currency, balance: number) {
        this.name = name;
        this.address = address;
        this.balance = balance;
        this.currency = currency;
        
        //TODO get state
        let rate = 10
        this.usdValue = balance * rate
        switch (currency) {
            case Currency.Polkadot:
                this.logo = require("assets/img/dot.png")
                this.symbol = "DOT"
                break
            case Currency.Kusama:
                this.logo = require("assets/img/kusama.png")
                this.symbol = "KSM"
                break
        }

    }
}
