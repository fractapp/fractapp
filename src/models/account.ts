import { Currency } from "./wallet";

/**
 * @category Models
 */
export class Account {
    name: string;
    address: string;
    pubKey: string;
    currency: Currency;
    balance: number;

    constructor(name: string, address: string, pubKey: string, currency: Currency, balance: number) {
        this.name = name
        this.address = address
        this.pubKey = pubKey
        this.currency = currency
        this.balance = balance
    }

    static parse(json: string): Account {
        return JSON.parse(json)
    }
}