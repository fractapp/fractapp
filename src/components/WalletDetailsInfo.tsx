import React, { useState } from 'react';
import { StyleSheet, View, Image, Text } from 'react-native';
import { Wallet } from 'models/wallet'

const WalletDetailsInfo = ({ account }: { account: Wallet }) => {
    return (
        <View style={styles.accountInfo}>
            <View style={{ flex: 1, flexDirection: 'row' }}>
                <Image
                    source={account.logo}
                    style={styles.logo}
                />
                <View style={{ flex: 1, flexDirection: 'column-reverse', marginLeft: 20 }}>
                    <View style={{ flexDirection: 'row', paddingTop: 2 }} >
                        <View style={{ flex: 1 }}>
                            <Text style={[styles.balance, { textAlign: "left" }]}>{account.balance} {account.symbol}</Text>
                        </View>
                        <View style={{ flex: 1 }}>
                            <Text style={[styles.balance, { textAlign: "right" }]}>{account.usdValue} $</Text>
                        </View>
                    </View>
                    <Text style={styles.name}>{account.name}</Text>
                </View>
            </View>
        </View>
    );
}
export default WalletDetailsInfo

const styles = StyleSheet.create({
    accountInfo: {
        marginTop: 15,
        alignSelf: 'center',
        width: "85%",
    },
    name: {
        fontSize: 16,
        fontFamily: "Roboto-Regular",
        fontStyle: "normal",
        fontWeight: "normal",
        color: "#888888"
    },
    balance: {
        fontSize: 25,
        fontFamily: "Roboto-Regular",
        fontStyle: "normal",
        color: "black"
    },
    logo: {
        width: 60,
        height: 60,
        borderRadius: 75,
        overflow: "hidden"
    },
});