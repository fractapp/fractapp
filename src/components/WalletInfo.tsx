import React from 'react';
import {StyleSheet, Text, View, TouchableHighlight, Image} from 'react-native';
import {getLogo, getSymbol, Wallet} from 'models/wallet';

/**
 * Component with wallet information
 * @category Components
 */
export const WalletInfo = ({
  wallet,
  onPress,
}: {
  wallet: Wallet;
  onPress: () => void;
}) => {
  const symbol = getSymbol(wallet.currency);
  return (
    <View style={{width: '100%', justifyContent: 'center'}}>
      <TouchableHighlight
        key={symbol}
        onPress={onPress}
        underlayColor="#f8f9fb">
        <View style={styles.account}>
          <View style={{flex: 1, flexDirection: 'row'}}>
            <Image source={getLogo(wallet.currency)} style={styles.logo} />
            <View
              style={{
                flex: 1,
                flexDirection: 'column-reverse',
                marginLeft: 20,
              }}>
              <View style={{height: 25, flexDirection: 'row'}}>
                <View style={{flex: 1}}>
                  <Text style={[styles.balance, {textAlign: 'left'}]}>
                    {wallet.balance} {symbol}
                  </Text>
                </View>
                <View style={{flex: 1}}>
                  <Text style={[styles.balance, {textAlign: 'right'}]}>
                    {wallet.usdValue}$
                  </Text>
                </View>
              </View>
              <Text style={styles.accountName}>{wallet.name}</Text>
            </View>
          </View>
        </View>
      </TouchableHighlight>
    </View>
  );
};

const styles = StyleSheet.create({
  account: {
    height: 50,
    paddingLeft: '5%',
    width: '95%',
    marginTop: 10,
    marginBottom: 10,
  },
  accountName: {
    fontSize: 15,
    fontFamily: 'Roboto-Regular',
    fontStyle: 'normal',
    fontWeight: 'normal',
    lineHeight: 18,
    color: 'black',
  },
  balance: {
    fontSize: 15,
    fontFamily: 'Roboto-Bold',
    fontStyle: 'normal',
    lineHeight: 18,
    color: 'black',
  },
  logo: {
    width: 50,
    height: 50,
    borderRadius: 75,
    overflow: 'hidden',
  },
});
