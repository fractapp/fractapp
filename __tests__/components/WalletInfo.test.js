import React from 'react';
import renderer from 'react-test-renderer';
import {WalletInfo} from 'components/WalletInfo';
import {Currency, Wallet} from 'types/wallet';
import StringUtils from 'utils/string';
import { AccountType, Network } from 'types/account';
import { WalletDetailsInfo } from 'components/WalletDetailsInfo';

jest.mock('react-native-i18n', () => ({
  t: (value) => value,
}));

it('Test Polkadot wallet', () => {
  const tree = renderer
    .create(
      <WalletInfo
        account={{
          name: 'Wallet Polkadot',
          address: 'address#1',
          pubKey:  'pubkey#1',
          currency: Currency.DOT,
          network: Network.Polkadot,
          viewBalance: 100,
          balance: {
            total:  '1000000000000',
            transferable:  '1000000000000',
            payableForFee:  '1000000000000',
          },
          type: AccountType.Main,
        }}
        price={100}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
