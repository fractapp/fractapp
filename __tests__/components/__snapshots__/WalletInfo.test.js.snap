// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Kusama wallet 1`] = `
<View
  style={
    Object {
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        Object {
          "height": 50,
          "marginBottom": 10,
          "marginTop": 10,
          "paddingLeft": "5%",
          "width": "95%",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../assets/img/kusama.png",
            }
          }
          style={
            Object {
              "borderColor": "black",
              "borderRadius": 25,
              "borderWidth": 0.8,
              "height": 50,
              "overflow": "hidden",
              "width": 50,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column-reverse",
              "marginLeft": 20,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "height": 25,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "black",
                      "fontFamily": "Roboto-Bold",
                      "fontSize": 15,
                      "fontStyle": "normal",
                      "lineHeight": 18,
                    },
                    Object {
                      "textAlign": "left",
                    },
                  ]
                }
              >
                20000000000000
                 
                KSM
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "black",
                      "fontFamily": "Roboto-Bold",
                      "fontSize": 15,
                      "fontStyle": "normal",
                      "lineHeight": 18,
                    },
                    Object {
                      "textAlign": "right",
                    },
                  ]
                }
              >
                $
                NaN
              </Text>
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "black",
                "fontFamily": "Roboto-Regular",
                "fontSize": 15,
                "fontStyle": "normal",
                "fontWeight": "normal",
                "lineHeight": 18,
              }
            }
          >
            Wallet Kusama
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Test Polkadot wallet 1`] = `
<View
  style={
    Object {
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        Object {
          "height": 50,
          "marginBottom": 10,
          "marginTop": 10,
          "paddingLeft": "5%",
          "width": "95%",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../assets/img/dot.png",
            }
          }
          style={
            Object {
              "borderColor": "black",
              "borderRadius": 25,
              "borderWidth": 0.8,
              "height": 50,
              "overflow": "hidden",
              "width": 50,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column-reverse",
              "marginLeft": 20,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "height": 25,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "black",
                      "fontFamily": "Roboto-Bold",
                      "fontSize": 15,
                      "fontStyle": "normal",
                      "lineHeight": 18,
                    },
                    Object {
                      "textAlign": "left",
                    },
                  ]
                }
              >
                1000000000000
                 
                DOT
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "black",
                      "fontFamily": "Roboto-Bold",
                      "fontSize": 15,
                      "fontStyle": "normal",
                      "lineHeight": 18,
                    },
                    Object {
                      "textAlign": "right",
                    },
                  ]
                }
              >
                $
                NaN
              </Text>
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "black",
                "fontFamily": "Roboto-Regular",
                "fontSize": 15,
                "fontStyle": "normal",
                "fontWeight": "normal",
                "lineHeight": 18,
              }
            }
          >
            Wallet Polkadot
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
